tests/unit/test_engine.py

```diff
@@ -598,6 +598,24 @@ def test_query(engine):
     assert model_query.index is None
 
 
+def test_query_subclass(engine):
+    """Engine.query supports model and index-based queries"""
+    class Admin(User):
+        pass
+
+    index_query = engine.query(
+        Admin.by_email,
+        key=Admin.by_email.hash_key == "placeholder",
+        forward=False
+    )
+    assert index_query.model is Admin
+    assert index_query.index is Admin.by_email
+
+    model_query = engine.query(Admin, key=User.Meta.hash_key == "other")
```

Comment:
Should this be allowed?  We could check that the condition's model is the query model, or (as it exists now) allow parent models to be used.  The former is a tight validation on the types, the latter is a tight validation on the names (since `User.Meta.hash_key` has the same name as `Admin.Meta.hash_key`).  Note that this may not hold if the subclass changes the hash key column.